Assignment1/src/test/java/files/ResourceTests.java

```diff
@@ -1,118 +1,109 @@
 package files;
-import static io.restassured.RestAssured.*;
 
-import java.util.ArrayList;
-import java.util.List;
+import static io.restassured.RestAssured.given;
 
-import org.apache.http.HttpRequest;
 import org.hamcrest.Matchers;
 import org.junit.Assert;
 import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
-import org.testng.asserts.SoftAssert;
 
-import io.cucumber.core.gherkin.messages.internal.gherkin.internal.com.eclipsesource.json.JsonArray;
-import io.restassured.RestAssured;
 import io.restassured.builder.RequestSpecBuilder;
 import io.restassured.builder.ResponseSpecBuilder;
 import io.restassured.http.ContentType;
-import io.restassured.parsing.Parser;
-import io.restassured.path.json.JsonPath;
-import io.restassured.response.Response;
 import io.restassured.specification.RequestSpecification;
 import io.restassured.specification.ResponseSpecification;
 
 public class ResourceTests {
-	RequestSpecification req = new RequestSpecBuilder().setBaseUri("https://reqres.in/").setContentType(ContentType.JSON).build();
-	ResponseSpecification res200 = new ResponseSpecBuilder().expectStatusCode(200).expectContentType(ContentType.JSON).build();
-	ResponseSpecification res404 = new ResponseSpecBuilder().expectStatusCode(404).expectContentType(ContentType.JSON).build();
-		
-	@Test(dataProvider = "validIds" )
+	RequestSpecification req = new RequestSpecBuilder().setBaseUri("https://reqres.in/")
+			.setContentType(ContentType.JSON).build();
+	ResponseSpecification res200 = new ResponseSpecBuilder().expectStatusCode(200).expectContentType(ContentType.JSON)
+			.build();
+	ResponseSpecification res404 = new ResponseSpecBuilder().expectStatusCode(404).expectContentType(ContentType.JSON)
+			.build();
+
+	@Test(dataProvider = "validIds")
 	private void test_get_singleResource_withValidId_returns_HTTP_200_UStory5(int validId) {
```

Comment:
It isn't a good idea to add user story number in your function names. This is because if the user story order changes or some more a added in between in the list, you will have to come back and change the name. More often you don't come back to change and then the function names are out of sync with the user story numbers and as a result it causes more confusion than add information.

Can you have a read about what is the right convention for naming functions? I think even for automated tests it should be `camelCase` and not `snake_case`. Ensuring such consistency reduces the mental overhead when reading code. One insight I learned over several years is that a piece of code is written once but read and modified tens or hundreds of times. So it is important to optimize the code for readability.

Reply:
Removed UserStory references.